### ISSUE TITLE
Issue with COM between PIC and Unity

### ISSUE THREAD

**Issue by ABoyOnFire** [dwilches/Ardity#30]
Hello dwilches,

Thank you for sharing this code base; and sample projects. 

I am pretty amateur with Unity/C#, and believe I have having an issue similar to that which **jeongcatfish** or **Andrea-Loriedo** seem to be describing. I did not really mess with the scene, or the code; so assume the Listener is per-configured in a mode for debugging. 

I was trying to use the **ReadWrite** sample scene but was not getting the expected behavior @ 115200 Baud rate. So I set up a quick test project using a 9600 Baud, and attempting to use the **DemoScene_AutoPoll** scene as you suggested this thread: (https://github.com/dwilches/Ardity/issues/19)

//Code above just configures the EUSART and I/Os which are not being used.
// System is running at 4 MHz Default. 
Below is the main code snippet:

while (1)
{
// Add your application code
     DELAY_milliseconds(1000);
     printf("Test Message\r\n");
}

![image](https://user-images.githubusercontent.com/49578271/67744305-5a9ff880-f9de-11e9-8656-f15dbbfb2d23.png)
![image](https://user-images.githubusercontent.com/49578271/67744238-37754900-f9de-11e9-9142-79e38fc7c421.png)
![image](https://user-images.githubusercontent.com/49578271/67744275-4b20af80-f9de-11e9-8957-39e1c5735704.png)
![image](https://user-images.githubusercontent.com/49578271/67744430-bbc7cc00-f9de-11e9-8a47-24c3779c95a0.png)

I can provide the packaged project, but it is basically just configuring a PIC device's EUSART.
It is POLLING driven, not doing any reads and there are no other task running besides the delay. DELAY_ is blocking; but since I am not doing any reading response yet, I would not expect that to have been an issue. 
I cloned from the latest HEAD. I don't have an Arduino for evaluation.

I can keep looking into this; but figured I would message to see if there was something simple being missed. 

Thank you for any assistance.
Best Wishes,
ABoyOnFire

**Comment by dwilches** [comment]
Hello!
Sure, please send me the compressed package, I think that'll make it easier to find what is going on.
You can send it through Google Drive or something similar.
Also, if you have the part of the code that runs in the Arduino, I can use that.
Cheers!

**Comment by ABoyOnFire** [comment]
Hello dwilches.
Thank you for the quick response. I found the issue, it was not related to the Unity code; that worked perfectly.

The USB-Serial Bridge device which doubles as the programmer seems to be underline issue. I believe there something with the flow control configuration effecting the communication.

I brought out the Saleae Logic Analyzer, and tested using the STM32, MCP2221, and a usb-uart-3-click (CP2102N) and was able to bridge the data as expected. Tested using the Polling and ReadWrite example projects. All worked like a champ.

Excited to start digging deeper into interfacing this with some hardware over the weekend. Great library, I'll shoot you a message if I make something neat with it. 

Cheers to you!
